### PR TITLE
PCC01 GeoffRiley

### PR DESCRIPTION
**Pull request by GeoffRiley**
Difficulty level (1-10): [2]
Estimated time spent (hours): [1]
Completed (yes/no): [Yes]
I stretched my coding skills (if yes what did you learn?): [Using the `key` parameter with `max` was new to me (as far as I can remember!) on this challenge.]
Other feedback (what can we improve?): [The template and test files are missing from the ./01/ directory and so I copied them across from the master distribution.
I created symlinks to the data.py and dictionary.txt files to avoid adding large duplicate files into git… but I did that before seeing the instruction not to submit these files.  I attempted to remove them from the git archive, but it then started attempting to do merges and so I left it as it is.  Sorry about that.
I spent far longer faffing around with git than I did with the programming. :)]

### CODE REVIEW

**Comment by pmayd** [comment]
I think you have done something wrong if your 01 is missing the tests and files, because you should start from master: https://github.com/pybites/challenges/tree/master/01

Just fork the repository, create a branch from master and you are good to go

**Comment by bbelderbos** [comment]
Thanks @pmayd for reviewing. We actually tell users in our instructions to branch off of community. I'd need to check if we lost templates files there ...

**Comment by bbelderbos** [comment]
Seems we lost them so:

```commit 629f6c927356f917f33db395498fc14eec145418
Author: Bob Belderbos <bobbelderbos@gmail.com>
Date:   Thu Sep 26 08:06:27 2019 +0200

    restored template files```

**Comment by pmayd** [comment]
Ah sorry for the misleading information

File: 01/GeoffRiley/wordvalue.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+def load_words():
+    """Load dictionary into a list and return list"""
+    with open(DICTIONARY) as f:
+        return [word for word in f.read().splitlines()]
+
+
+def calc_word_value(word):
+    """Calculate the value of the word entered into function
+    using imported constant mapping LETTER_SCORES"""
+    return sum([LETTER_SCORES[x.upper()] for x in word if x.isalpha()])
```
[the remark this answers is not in the document]
Comment (GeoffRiley) [reply 01/GeoffRiley/wordvalue.py]: I knew there should be a way to get the value without having to do a specific check, I just couldn't bring it to mind and didn't happen upon it when I went a'googling.  Thanks, I'll endeavour to remember that one.

File: 01/GeoffRiley/wordvalue.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def max_word_value(words=None):
+    """Calculate the word with the max value, can receive a list
+    of words as arg, if none provided uses default DICTIONARY"""
+    if words is None:
```
[the remark this answers is not in the document]
Comment (GeoffRiley) [reply 01/GeoffRiley/wordvalue.py]: Darn, I had split it whilst debugging and forgot to join it back together again!  I agree completely. :)

File: 01/GeoffRiley/wordvalue.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        words = load_words()
+    words_values = {x: calc_word_value(x) for x in words}
+    return max(words_values, key=lambda k:words_values[k])
```
[the remark this answers is not in the document]
Comment (GeoffRiley) [reply 01/GeoffRiley/wordvalue.py]: I went through a number of iterations on this chunk of code, I tried to do it in a single line, but always ended up with something getting repeated… in the 'Agile' parlance, I try to keep things DRY.  This seemed like the best effort.

Thank you for your critique, I appreciate it very much.

[the remark this answers is not in the document]
Comment (pmayd) [reply 01/GeoffRiley/wordvalue.py]: You are more than welcome ;)

**Pull request merged**